test/integration/integration_test.go

```diff
@@ -316,6 +316,47 @@ func TestArchiveAndRead(t *testing.T) {
 	if retryErr != nil {
 		t.Fatal(retryErr)
 	}
+
+	// Query cronjob log urls
+	url = fmt.Sprintf("https://localhost:8081/apis/batch/v1/namespaces/%s/cronjobs/generate-log-1/log", namespaceName)
+	request, errRequest = http.NewRequest("GET", url, nil)
+	if errRequest != nil {
+		fmt.Printf("could not create a request, %s", errRequest)
+		t.Fatal(errRequest)
+	}
+	request.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token.Status.Token))
+	fmt.Printf("HTTP request: %s\n", request.URL.RequestURI())
+
+	retryErr = retry.Do(func() error {
+		response, errResp := clientHTTP.Do(request)
+		if errResp != nil {
+			fmt.Printf("Couldn't get a response HTTP, %s\n", errResp)
+			return errResp
+		}
+		defer response.Body.Close()
+
+		body, errReadAll := io.ReadAll(response.Body)
+		if errReadAll != nil {
+			fmt.Printf("Couldn't read Body, %s\n", errReadAll)
+			return errReadAll
+		}
+
+		if response.StatusCode != http.StatusOK {
+			fmt.Printf("The HTTP status returned is not OK, %s\n", response.Status)
+			return errors.New(response.Status)
+		}
+		var logUrls []string
+		json.Unmarshal([]byte(body), &logUrls)
+		if len(logUrls) < 1 {
+			return errors.New("could not retrieve the cronjob pod log urls")
+		}
```

Comment:
This is not checking properly what it is suposed to be. I'm printing the logUrls and its an empty slice. I guess that there is an error and its not handled, as Andrea suggested.

Reply:
One solution I got:

```suggestion
		type Response struct {
			URLs []string `json:"-"`
		}
		var data Response
		err := json.Unmarshal(body, &data)
		if err != nil {
			return err
		}
		if len(data.URLs) < 1 {
			return ...
	    }
```

Reply:
You can use an anonymous struct instead:

```
logUrls := struct{ URLs []string }{}
```

I'm not sure why `[]string` is not working as expected.

Reply:
the urls are empty because they are not configured so the urls in the log_url tables are empty... I could add some configuration to also test how the sink introduce the urls but it's not what I did

Reply:
Hm... then just ensure that the endpoint does not return 404. Now its misleading, and its not properly coded. It tells "if there are less than one log result then fail, if there are one or more then the test is successful"

Reply:
How could you ever have a negative number here?  Shouldn't it just be `== 0`?

Reply:
> How could you ever have a negative number here? Shouldn't it just be == 0?

yes, I will change it, I just didn't put too much thought on it, but I agree it's more readable `== 0`  as "if it's empty" than `< 1` as "it should have at least one"

Reply:
@ggallen I've made the change to `==0`

@rh-hemartin I've included the config of logging to the sink, so now the urls aren't empty